form_process.php

```diff
@@ -1,6 +1,9 @@
 <?php
 
 	// only Ajax requests come through here
+	if (!session_id() || session_id() == '') {
```

Comment:
No need for the second condition -- they test for the same thing, once you factor in type coercion.